### PR TITLE
adds spawning decay datasets for updating decay logic

### PR DESCRIPTION
**Pull request by ergz**
this PR adds data resulting from the updated spawning decay analysis. Note that the 130 files changed is moistly the result of adding all of the rating curves from the srh2d model into the data-raw folder. The most important files to review are the cache-habitat and data-raw/spawning-habitat-decay/spawning-decay.R

### CODE REVIEW

File: R/data.R [hunk 1]
```diff
@@ -946,3 +947,38 @@ NULL
...
+"gravel_size_to_prop_of_movement"
+
+#' @title Watershed Decay Status
+#' @description Watershed to TRUE/FALSE lookup indicating whether habitat decay occurs.
+"watershed_decay_status"
+
+#' @title Watershed Spawning Habitat Decays
+#' @description The total decay by watershed. This includes watersheds with no decay, these simply
+#' have all values equal to 0. For watersheds with decay, there are three different curves,
+#' 'min', 'avg', 'max'.
```
Comment (ErinCain) [inline R/data.R:961]: Is there a source for where these decay values come from?

File: data-raw/cached-habitat.R [hunk 2]
```diff
@@ -944,3 +944,33 @@ usethis::use_data(delta_contact_points, overwrite = TRUE)
...
+  "Stanislaus River",
+  "Mokelumne River", 
+  "Tuolumne River"
+) |> setNames(fallRunDSM::watershed_labels)
+
+usethis::use_data(watershed_decay_status, overwrite = TRUE)
+
+
+# average spawning habitat
+spawning_habitat_average <-
```
Comment (ErinCain) [inline data-raw/cached-habitat.R:967]: Do we want to have lfr in here too for once we we have it finalized and versioned? Not needed now but might be worth adding so we do not have to do it later.

File: data-raw/spawning-habitat-decay/spawning-decay.R [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+    sed_ft3_per_second_avg = mean(sed_ft3_per_second_avg),
+    sed_ft3_per_second_max = mean(sed_ft3_per_second_max),
+    sed_ft3_per_day_min = mean(sed_ft3_per_day_min), 
+    sed_ft3_per_day_avg = mean(sed_ft3_per_day_avg),
+    sed_ft3_per_day_max = mean(sed_ft3_per_day_max),
+  ) |>
+  ungroup() |> 
+  mutate(flow_cfd = flow_cfs * 86400)
+
+rating_curve_for_calib <- rating_curve
```
Comment (ErinCain) [inline data-raw/spawning-habitat-decay/spawning-decay.R]: Any reason why we have 2 different objects - rating_curv_for_calib and rating_curve? It seems fine in the use case you have but I wouldn't want to update one and forget to update the other if they should be the same.

File: data-raw/spawning-habitat-decay/spawning-decay.R [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+    date = kwk_usgs$Date, 
+    flow = kwk_usgs$Flow, 
+    sediment_transport_f3_day = calib_sed_curve(flow)
+  ) |> 
+    mutate(sediment_transport_f3_day = ifelse(is.na(sediment_transport_f3_day), 0, sediment_transport_f3_day))
+  
+  
+  calib_kwk_sed_transport_sim <- calib_kwk_sed_transport |> 
+    filter(date >= "2015-01-01", date <= "2017-04-01") |> 
+    mutate(sediment_transport_f3_day = ifelse(is.na(sediment_transport_f3_day), 0, sediment_transport_f3_day),
```
Comment (ErinCain) [inline data-raw/spawning-habitat-decay/spawning-decay.R]: Shouldn't this already be 0 - looks like NA were replaced by 0 in lines 136 - 137 above. Not an issues but may be duplicative.

File: NEWS.md [hunk 5]
```diff
@@ -1,3 +1,20 @@
...
+well as a new script under `data-raw/spawning-habitat-decay` that described the new habitat calculations. The following data has been added along with correspondings documentations:
+    - `DSMhabitat::watershed_spawning_decays` - dataframes with raw decay for three calcualted curves
+    - `DSMhabitat::watershed_decay_status` - a convinience dataframe that shows whether a given watershed has spawning decay or not.
+    - `DSMhabitat::spawning_habitat_average` - a convinience dataset with average spawning habitat by watershed and run.
+    - `DSMhabitat::spawning_decay_multiplier` - a datset cotaining 31 watershed decay multipliers. Currently this cotains only Fall-run values.
+
+* **CALSIM updates**. Update DSMhabitat datasets integrating in new 2019 biop flows. The result of
+doing so means each dataset that is generated using flow as an input will have the following form: 
+`fr_fp$biop_2008_2009` and `fr_fp$biop_itp_2018_2019`.
+
```
Comment (ErinCain) [inline NEWS.md:16]: Do we want to add a: 

Habitat related datasets for use with the following DSM models:
* [Fall Run DSM (v3.0)](https://github.com/CVPIA-OSC/fallRunDSM/releases/tag/v2.0)
* [Late Fall Run DSM (v3.0)](https://github.com/CVPIA-OSC/fallRunDSM/releases/tag/v1.0)
* [Winter Run DSM (v3.0)](https://github.com/CVPIA-OSC/winterRunDSM/releases/tag/v2.0)
* [Spring Run DSM (v3.0)](https://github.com/CVPIA-OSC/springRunDSM/releases/tag/v2.0)

File: data-raw/spawning-habitat-decay/spawning-decay.R [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+  geom_line(aes(date, current_vol)) + 
+  labs(x = "", y = "Spawning Habitat Vol. (cubic feet)") + 
+  theme_bw()
+
+p2 <- kwk_usgs |> 
+  filter(Date >= "2015-01-01", Date <= "2017-04-01") |> ggplot(aes(Date, Flow)) + geom_line() + 
+  theme_bw()
+
+# plot to confirm scaledown did the right thing
+grid.arrange(p2, p1, nrow = 2)
```
Comment (ErinCain) [inline data-raw/spawning-habitat-decay/spawning-decay.R:192]: Plot looks good and helps me understand how the decay is working!

File: data-raw/spawning-habitat-decay/spawning-decay.R [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+  }
+}) |> 
+  set_names(fallRunDSM::watershed_labels)
+
+usethis::use_data(watershed_spawning_decays, overwrite = TRUE)
+
+watershed_decays$`American River` |> 
+  ggplot(aes(date, decay_acres_month, color = decay_type)) + geom_line()
+
+watershed_decays$`American River` |>
```
Comment (ErinCain) [inline data-raw/spawning-habitat-decay/spawning-decay.R]: When running the script I can't find the object `watershed_decays`. I think you may have renamed it watershed_spawning_decays.

File: data-raw/spawning-habitat-decay/spawning-decay.R [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+  `Cosumnes River` = "none", `Mokelumne River` = "avg", `Merced River` = "avg", 
+  `Stanislaus River` = "avg", `Tuolumne River` = "min", `San Joaquin River` = "none"
+)
+
+
+watersheds_with_decay <- names(which(DSMhabitat::watershed_decay_status))
+
+spawning_decay_multiplier <- purrr::map(names(watershed_decay_level_lookups), function(w) {
+  if (w %in% watersheds_with_decay) {
+    decay <- DSMhabitat::watershed_decays[[w]] |>
```
Comment (ErinCain) [inline data-raw/spawning-habitat-decay/spawning-decay.R]: same comment as above. Was not able to run this function

**Review by ErinCain — COMMENTED** [review]
Looks good. One change where I wasn't able to find a data object that I think you renamed. Other than that just a few clarifying comments but looked good!

File: R/data.R (hunk 1 above)
Comment (ergz) [reply R/data.R:961]: for this I was thinking of converting the spawning decay analysis script into a vignette with details as to how the curves are calculated, and pointing the docs to this vignette. This doc would become an article in the pkgdown site, I started the vignette, should be done early next week

File: data-raw/spawning-habitat-decay/spawning-decay.R (hunk 3 above)
Comment (ergz) [reply data-raw/spawning-habitat-decay/spawning-decay.R]: ooof, a hacky thing I did on the original script that got copied over....fixing

File: data-raw/spawning-habitat-decay/spawning-decay.R (hunk 8 above)
Comment (ergz) [reply data-raw/spawning-habitat-decay/spawning-decay.R]: fixed

File: data-raw/spawning-habitat-decay/spawning-decay.R (hunk 7 above)
Comment (ergz) [reply data-raw/spawning-habitat-decay/spawning-decay.R]: fixed

File: NEWS.md (hunk 5 above)
Comment (ergz) [reply NEWS.md:16]: I think we should, it doesn't have to be part of this PR tho

**Review by ErinCain — APPROVED** [review]
Thanks for making thoes changes! Looks great

**Pull request merged**